GeometryCore/src/Mesh.cpp

```diff
@@ -164,4 +165,47 @@ namespace geom_utils
 		return true;
 	}
 
+	Mesh createFoundation(const Mesh& inputModel, const FPoint3D::coord foundationHeight, const FPoint3D::coord inflateValue) {
+		auto from2dTo3dTriangle = [](const Triangle2D& input_triangle) { return Triangle3D(FPoint3D(input_triangle.a.x, input_triangle.a.y, 0), FPoint3D(input_triangle.b.x, input_triangle.b.y, 0), FPoint3D(input_triangle.c.x, input_triangle.c.y, 0)); };
+		auto moveUp = [](std::vector<Triangle3D>& foundation_part, FPoint3D::coord by) {for_each(foundation_part.begin(), foundation_part.end(), [&](Triangle3D& input) {input.a.z += by; input.b.z += by; input.c.z += by; }); };
+
+		std::vector<Triangle3D> facets(inputModel.getFacets());
+		Polygon verticles;
+
+		for (auto i : facets) {
+			verticles.add(FPoint2D(i.a.x, i.a.y));
+			verticles.add(FPoint2D(i.b.x, i.b.y));
+			verticles.add(FPoint2D(i.c.x, i.c.y));
+		}
+		auto convex = verticles.convexHull();
+		convex.simplify(convex.polygonLength() * 0.01);
+		auto plane_for_foundation = convex.inflate(inflateValue);
+
+		auto polygon_of_bottom = plane_for_foundation.triangulate();
+
+		std::vector<Triangle3D> foundation_bottom;
+		std::for_each(polygon_of_bottom.begin(), polygon_of_bottom.end(), [&](const Triangle2D& input_triangle) {foundation_bottom.push_back(from2dTo3dTriangle(input_triangle)); });
```

Comment:
Actually, I meant apply camelCase for all varianbles, not just for `from2dTo3dTriangle`.
Also, want to show you the next trick. You can use this kind of idiom in case if you need transform some data set into some other representation (in our case array of 2d triangles into 3d):
`std::transform(polygon_of_bottom.begin(), polygon_of_bottom.end(), std::back_inserter(foundation_bottom), from2dTo3dTriangle);`

Reply:
This trick is really useful, this more readable than using lambdas and for_each